OracleGoldenGate/21c/install-deployment.sh

```diff
@@ -44,6 +44,26 @@ function ogg_installer_setup() {
     unzip -q "${INSTALLER}" -d "/tmp/installer" || abort "Unzip operation failed for '${INSTALLER}'"
     chmod -R o=g-w             "/tmp/installer"
 }
+##
+##
+## o g g _ i n s t a l l _ o p t i o n
+##
+function ogg_install_option() {
+    # Get the path to the fastcopy.xml file that contains the metadata for the install option.
+    local fast_copy_file
+    fast_copy_file=$(find /tmp/installer -name fastcopy.xml)
+
+    # Get the xml line that needs to be parsed.
+    local xml_line
+    xml_line=$(grep -h 'TOPLEVEL_COMPONENT NAME' "${fast_copy_file}")
+
+    # Match string example:   <TOPLEVEL_COMPONENT NAME="oracle.oggcore.services" INSTALL_TYPE="ora21c" PLATFORM="Linux">
+    local regex
+    regex='.*<TOPLEVEL_COMPONENT NAME=.* INSTALL_TYPE="(.*)" .*'
+
+    [[ $xml_line =~ $regex ]]                     || abort "Could not find INSTALL_TYPE in the file '${fast_copy_file}'"
+    echo "${BASH_REMATCH[1]}"
+}
```

Comment:
This function is not very succinct. A more efficient implementation is:

```.sh
function ogg_install_option() {
    awk '/INSTALL_TYPE/ {
            sub(/^.*INSTALL_TYPE="/, "")   
            sub(/".*/, "")
            print
            exit 0
        }' $(find /tmp/installer -name fastcopy.xml -print -quit)
}
```

Reply:
Thanks! Steve, this looks really good.

There is one issue with this function, in case the `find` command does not return any output, `awk` will just wait there indefinitely expecting an input file.  I will try to address that.

Reply:
Btw, another implementation similar to yours using `perl` which IMO is more succinct. 


```bash
function ogg_install_option() {
    local fastcopy_file
    fastcopy_file=$(find /tmp/installer -name fastcopy.xml -print -quit)
    [[ -z "${fastcopy_file}" ]] && abort "The file 'fastcopy.xml' could not be located."
    perl -ne '/^.* INSTALL_TYPE="(.*)" .*$/ && print $1' "${fastcopy_file}"
}
```

Reply:
We never, ever use Perl.